EIPS/EIPs/eip-5075.md

```diff
@@ -0,0 +1,236 @@
+    }
+    //removes whitelist addresses as the msg.sender to not be rate limited
+    function removeWhitelist(address addrs) public {
+        require(msg.sender == authAddress);
+        whitelisted[addrs] = 0;
+    }
+
+    //gets the token amount left to be within the allowable limit
+    function getLimitLeft(address token)
+        public
+        view
+        returns (uint256 rateLeft)
+    {
+        TokenInfo storage info = infoL[token];
+        uint256 rate;
+        //if outside the time dwindow
+        if (timeLimit <= block.timestamp - info.timestamp) {
+            return (0);
+        }
+        //if the last transaction was within the time window, decreases the tracked outflow rate relative to the time elapsed, so that the limit is able to update in realtime rather than in blocks, making flows smooth, and increasing the rate available as time increases without a transaction
+        else {
+            rate = info.rate;
+            uint256 limitUnlocked = uint224(
+                (address(this).balance * rateLimit) /
+                    (timeLimit / (block.timestamp - info.timestamp)) /
+                    10000
+            );
+            if (rate <= limitUnlocked) {
+                return 0;
+            } else {
+                return rate - limitUnlocked;
+            }
+        }
+    }
+
+    //used to replace ERC20 erc20token.transfer() and ETH address.transfer() fns in all public accessible fns which change the balances for the contract as outflow transactions.
+    //to used for the recipient address, amount for value amount in raw value, token as the tokens contract address to check, for ETH use address(0x0)
+    function transferL(
+        address to,
+        uint256 amount,
+        address token
+    ) public returns (bool success) {
+        TokenInfo storage info = infoL[token];
+
+        if (whitelisted[msg.sender] == 1) {
+            if (address(token) == address(0)) {
+                payable(to).transfer(amount);
+                return(true);
+            } else {
+                return(IERC20(token).transfer(to, amount));
+            }
+        }
+        //used if the asset is ETH and not an ERC20 token
+        else {
+            if (address(token) == address(0)) {
+                //used to get around solidity 0.8 reverts
+                unchecked {
+                    //checks to see if the last transaction was outside the time window to track outflow for limiting
+                    if (timeLimit <= block.timestamp - info.timestamp) {
+                        info.rate = 0;
+                    }
+                    //if the last transaction was within the time window, decreases the tracked outflow rate relative to the time elapsed, so that the limit is able to update in realtime rather than in blocks, making flows smooth, and increasing the rate available as time increases without a transaction
+                    else {
+                        info.rate -= uint224(
```

Comment:
I think this may revert due to underflow if balance increased by a lot since the last rate was recorded, or if rateLimit or timeLimit paremeters changed in a way to case underflow. Should probably check for this before trying to subtract.

Reply:
Will do